### ISSUE TITLE
Ubuntu 17.10 RTL8821AE

### ISSUE THREAD

**Issue by Object49** [lwfinger/rtlwifi_new#283]
**Lenovo 300-15ibr Ubuntu 17.10 is a very weak signal WIFI even near the router, periodically the signal disappears altogether. Prior to this was Ubuntu 16.04 version, but the problem was there. On Windows everything was fine (I'm on linux only the second day).**
`~$ sudo lshw -C network`
-network
       description: Wireless interface
       Product: RTL8821AE 802.11ac PCIe Wireless Network Adapter
       manufacturer: Realtek Semiconductor Co., Ltd.
       physical ID: 0
       information about the bus: pci @ 0000: 02: 00.0
       logical name: wlp2s0
       version: 00
       Serial No.: 78: 45: 61: f0: 2d: 9d
       bit depth: 64 bits
       frequency: 33MHz
       Features: pm msi pciexpress bus_master cap_list ethernet hard wireless
       configuration: broadcast = yes driver = rtl8821ae driverversion = 4.13.0-16-generic firmware = N / A latency = 0 link = no multicast = yes wireless = IEEE 802.11
       resources: IRQ: 122 ioport: 2000 (size = 256) memory: 94200000-94203fff
`~$ rfkill list all`
0: ideapad_wlan: Wireless LAN
	Soft blocked: no
	Hard blocked: no
1: ideapad_bluetooth: Bluetooth
	Soft blocked: yes
	Hard blocked: no
2: phy0: Wireless LAN
	Soft blocked: no
	Hard blocked: no
3: hci0: Bluetooth
	Soft blocked: yes
	Hard blocked: no
`~$ uname -a`
Linux PC 4.13.0-16-generic #19-Ubuntu SMP Wed Oct 11 18:35:14 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
**I completed all the commands:**
`sudo apt-get install linux-headers-generic build-essential git`
`git clone http://github.com/lwfinger/rtlwifi_new.git`
`cd rtlwifi_new`
`make`
`sudo make install`
`sudo modprobe rtl8821ae`
**Here it can be seen that the drivers are placed from another card.**
![2017-10-23 13-55-51](https://user-images.githubusercontent.com/33023004/31885894-c970cfe6-b7fa-11e7-81d1-599401d42d21.png)
**And here the problem itself (the laptop is now one meter from the router).**
![2017-10-23 14-00-36](https://user-images.githubusercontent.com/33023004/31885999-2fac0c12-b7fb-11e7-90bf-b5aa7c394075.png)
**Help with this problem, I will be grateful!**

**Comment by Object49** [comment]
Sorry if the translation is not, I'm Russian.

**Comment by lwfinger** [comment]
Some vendors have put only one antenna into their laptops. In the Linux driver, the antenna connector specified in the EEPROM is selected. If the EEPROM coding is wrong, then it is not possible to have a strong signal. This problem does not show up on Windows because the vendor has modified the driver for that specific laptop. We have seen the problem with RTL8723BE devices, but there has not yet been a confirmation of the problem with RTL8821AE.

If possible, please verify that the laptop has only one antenna wire connected to the wireless card, and that moving that antenna lead to the other connector leads to a stronger signal for Linux. If both are true, we might be able to apply the same kind of fix that was done for RTL8723BE. Please tell me which connector (M, or A) had the antenna connection before you changed anything.

**Comment by Object49** [comment]
А где узнать про этот разъем?



>Понедельник, 23 октября 2017, 19:08 +03:00 от lwfinger <notifications@github.com>:
>
>Some vendors have put only one antenna into their laptops. In the Linux driver, the antenna connector specified in the EEPROM is selected. If the EEPROM coding is wrong, then it is not possible to have a strong signal. This problem does not show up on Windows because the vendor has modified the driver for that specific laptop. We have seen the problem with RTL8723BE devices, but there has not yet been a confirmation of the problem with RTL8821AE.
>If possible, please verify that the laptop has only one antenna wire connected to the wireless card, and that moving that antenna lead to the other connector leads to a stronger signal for Linux. If both are true, we might be able to apply the same kind of fix that was done for RTL8723BE. Please tell me which connector (M, or A) had the antenna connection before you changed anything.
>—
>You are receiving this because you authored the thread.
>Reply to this email directly,  view it on GitHub , or  mute the thread .


С уважением,
Денис Панасенко
meatbeard@mail.ru

**Comment by Object49** [comment]
And where can I find out about this connector?

**Comment by lwfinger** [comment]
It is on the wifi card. If you have only one antenna, it will be attached to A or M. If you have two antenna, both connectors will have a wire.

The wifi card will be somewhere inside the laptop. Most models have an access plate on the bottom.

**Comment by Object49** [comment]
@lwfinger
 **I will write more about my manipulations with the laptop:**
Before I put Ubuntu (it was 2 days ago), I had Windows 10 (I used this system all my life, with Linux it never mattered). One day, I left the laptop turned on for the night, turned on the film and went to bed. When I woke up there was no internet connection. I rebooted the modems, but the Internet did not appear because there was a fault of the provider. Before I go to make myself a tea, I put the laptop on a reboot. When I got back, I found that the laptop was hanging on the Lenovo logo. I started to surf the Internet in search of solutions, but I did not find it (I made many operations with the laptop). 
I needed files on the laptop so I did not want to reinstall the system without unloading files from there. Desperate, I wrote a question in Mail and there I was advised to download Ubuntu via LiveCD and throw off the files (with WIFI already there were problems). I did it. And while I was doing this, I liked Ubuntu and I decided not to return to Windows. Installed 16.04 LTS, but there were problems with WIFI. After turning over many pages of the Internet, I did not find a solution to the problem. I decided to upgrade to 17.10, but I did not find a solution. This is the story.
**I have knowledge of computers based on a normal user, so I do not understand the names of modules. Do I need to disassemble the laptop?**
Once again, I apologize for the translation.

**Comment by Object49** [comment]
**I am disassemble laptop. Many thanks for responding!**

**Comment by lwfinger** [comment]
OK, I understand. It seems that you will not want to go back to Windows. When you find the wifi card, take a picture and post it here.

**Comment by Object49** [comment]
**I found the problem. When I disassembled and did not even notice that it was not connected at all)))). I connected these wires and now the signal is complete. Thank you very much for your help !!!!**
I have a computer, it's on Windows, and on a laptop I'll be with Linux, I really like this system, but I'm a composer, and on Linux there's no FL Studio, so the PC will remain with Windows.
**Here's a photos just in case.**
![wifi1](https://user-images.githubusercontent.com/33023004/31915587-a5f73a8e-b857-11e7-821f-19b9ccda1cbd.jpeg)
![wifi2](https://user-images.githubusercontent.com/33023004/31915802-90e78f58-b858-11e7-939f-83a1cb2ccb44.jpeg)
![wi1](https://user-images.githubusercontent.com/33023004/31915861-cfcb7e6e-b858-11e7-913f-eac36deb0509.jpeg)
![wi2](https://user-images.githubusercontent.com/33023004/31915909-ff276c36-b858-11e7-85a1-f8ae11ac711c.jpeg)
